src/catch2/internal/catch_test_registry.hpp

```diff
@@ -47,6 +47,33 @@ Detail::unique_ptr<ITestInvoker> makeTestInvoker( void (C::*testAsMethod)() ) {
     return Detail::make_unique<TestInvokerAsMethod<C>>( testAsMethod );
 }
 
+template <typename C>
+class TestInvokerFixture : public ITestInvoker {
+    void ( C::*m_testAsMethod )();
+    Detail::unique_ptr<C> m_fixture = nullptr;
+
+public:
+    TestInvokerFixture( void ( C::*testAsMethod )() ) noexcept : m_testAsMethod( testAsMethod ) {}
+
+    void testCaseStarting() override { 
+        m_fixture = Detail::make_unique<C>(); 
+    }
+
+    void testCaseEnding() override { 
+        m_fixture.reset(); 
+    }
+
+    void invoke() const override {
+        auto* f = const_cast<C*>( m_fixture.get() );
+        ( f->*m_testAsMethod )();
+    }
+};
+
+template<typename C>
+Detail::unique_ptr<ITestInvoker> makeTestInvokerFixture( void ( C::*testAsMethod )() ) {
```

Comment:
I could not make a `makeTestInvoker` overload since it would have the same function signature as the factory function for TEST_CASE_METHOD.